File: leetcode/easy/118.Pascal's Triangle/python/pascalsTriangle.py
def generate(numRows):
    main_array = [[1]]
    for i in range(numRows - 1):
        temp_array = [0] + main_array[-1] + [0]
        sub_array = []
        for j in range(len(temp_array) - 1):
            new_element = temp_array[j] + temp_array[j + 1]
            sub_array.append(new_element)
        main_array.append(sub_array)
    return main_array


# Testing
test_data1 = 5
expected1 = [[1],[1,1],[1,2,1],[1,3,3,1],[1,4,6,4,1]]
print(generate(test_data1))
a1 = [1,2,3]
a2 = a1[:]
a1.pop()
print(a1)
print(a2)

File: leetcode/easy/118.Pascal's Triangle/js/pascalsTriangle.js
"use strict";
/**
 * @param {number} numRows
 * @return {number [][]}
 * @author Alex Soto
 */

var generate = function(numRows) {
    let mainArray = [[1]];
    for (let i = 0; i < numRows - 1; i++) {
        let tempArray = mainArray[mainArray.length - 1].slice()
        tempArray.unshift(0);
        tempArray.push(0);
        let subArray = [];
        for (let j = 0; j < tempArray.length - 1; j++) {
            let newElement = tempArray[j] + tempArray[j + 1];
            subArray[j] = newElement;
        };
        mainArray.push(subArray);
    };
    return mainArray;
};

// Testing
let numRow = 5;
let expected1 = [[1],[1,1],[1,2,1],[1,3,3,1],[1,4,6,4,1]];
console.log(generate(numRow));